test/permissionlessOrderBookReserveLister.js

```diff
@@ -260,22 +277,133 @@ contract('PermissionlessOrderBookReserveLister', async (accounts) => {
+        await kyberNetwork.setExpectedRate(expectedRate);
+        await kyberNetwork.setFeeBurner(feeBurner.address);
+        await kyberNetwork.setParams(gasPrice.valueOf(), negligibleRateDiff);
+        await kyberNetwork.addOperator(operator);
+        await kyberNetwork.setEnable(true);
+        await kyberNetwork.addOperator(lister.address);
+
+        // list an order book reserve
+        await lister.addOrderBookContract(someToken.address);
+        await lister.initOrderBookContract(someToken.address);
+        await lister.listOrderBookContract(someToken.address);
+
+        // add orders
+        const reserve = await OrderBookReserve.at(
+            await lister.reserves(someToken.address)
+        );
+        let amountTokenInWei = new BigNumber(500 * 10 ** 18); //500 tokens
+        let amountKncInWei = new BigNumber(600 * 10 ** 18);
+        let amountEthInWei = new BigNumber(2 * 10 ** 18);
+        await makerDeposit(
+            reserve,
+            maker,
+            amountEthInWei /* ethWei */,
+            amountTokenInWei /* tokenTwei */,
+            amountKncInWei /* kncTwei */,
+            kncToken
+        );
+
+        await reserve.submitBuyTokenOrder(
+            2 * 10 ** 18 /* srcAmount */,
+            100 * 10 ** 18 /* dstAmount */,
+            {from: maker}
+        );
+
+        // swap ETH to someToken
+        const ethWeiToSwap = 0.5 * 10 ** 18;
+        await kyberProxy.swapEtherToToken(
+            someToken.address /* token */,
+            1 /* minConversionRate */,
+            {from: taker, value: ethWeiToSwap}
+        );
+
+        // burn fees
+        const result = await feeBurner.burnReserveFees(reserve.address);
+
+        // Assert
+        const burnAssignedFeesEvent = result.logs[0];
+        burnAssignedFeesEvent.event.should.equal('BurnAssignedFees');
+        burnAssignedFeesEvent.args.reserve.should.equal(reserve.address);
+
+        // (ethWeiToSwap * (300: kncPerETHRate) * (25: BURN_FEE_BPS) / 10000) - 1
+        const expectedFeesInKncWei = BigNumber("374999999999999999");
+        burnAssignedFeesEvent.args.quantity.should.be.bignumber.equal(
+            expectedFeesInKncWei
+        );
+    });
+});
 
 function log(str) {
     console.log(str);
 }
 
-async function makerDeposit(res, maker, ethWei, tokenTwei, kncTwei) {
-
+async function makerDeposit(res, maker, ethWei, tokenTwei, kncTwei, kncToken) {
     await token.approve(res.address, tokenTwei);
-    await res.makerDepositToken(maker, tokenTwei);
-    await KNCToken.approve(res.address, kncTwei);
-    await res.makerDepositKnc(maker, kncTwei);
-    await res.makerDepositWei(maker, {from: maker, value: ethWei});
+    await res.depositToken(maker, tokenTwei);
+    await kncToken.approve(res.address, kncTwei);
+    await res.depositKncFee(maker, kncTwei);
+    await res.depositEther(maker, {from: maker, value: ethWei});
 }
 
 async function twoStringsSoliditySha(str1, str2) {
```

Comment:
seems can remove this function...
I guess I put it there. but if u make changes. remove it.

Reply:
Removed unused functions.